### DISCUSSION TITLE
Goblin auctioneer / Lottery system idea

### DISCUSSION THREAD

**Discussion by Telknub** [sunflower-land/sunflower-land#2130] [Ideas]
This is an idea I had with help of other brazilians in the portuguese chat to make the goblin auctioneer fair for everyone and less frustrating.

The main idea revolves around tickets, similar to how Adam said in #2122 , but with a few differences.

The goblin auctioneer would function in the same way Adam said:

>When an auction opens, players have 24 hours to cast their bids 

>Resources/SFL are escrowed (off chain) until the auction is over - players can not access these items. This guarantees that a player is actually intending to mint the item.

>After 24 hours, players come back to check the results of the auction.

This is much better than the current system we have, but it still has its problems, and with rng it still can be very frustrating since the way Adam explained it would rely on RNG.

To understand the full idea we need a few steps:

- First we need to align how players can get tickets:

The idea is, in the main island, we have a goblin NPC, with a board on his side. Every 24h he will give the player 3  quests, with 3 different tiers of difficulty. Even with different tiers of difficulty, the quests given by the goblin are not supposed to be difficult, it should be able to be completed by almost everyone. (the hardest quests should be something like mine 2 gold, collect 6 fruit, etc) 

Also the easier quests would give more tickets than the harder ones, so if a new player can't make the harder quest because he does not have 2 gold ores or fruits trees unlocked he would not fall too much behind more experienced players, while at the same time those more experienced players would be rewarded with a bit more tickets.

Example:

- The player opens the game and clicks on the goblin NPC 
- He sees 3 quests every day: an easy, a medium and a hard one
- The easy is something like "chop 5 trees" and rewards the player with 3 tickets
- the medium is something like "harvest 30 cauliflowers" and rewards the player with 2 tickets
- the hard one is "harvest 2 gold" and rewards the player with 1 ticket

These quests would be the same for all players, but every day they change to something different at random picked from a pool of possible quests, creating variety on what players need to do.
With a suggestion in discord from Azaro and GodIsGreat these quests while being the same for everyone on each day, they could scale based of the amount of resources on your island, so for example, instead of "chop 5 trees" it can be something like "chop 2*number of trees of the player".
This quests could also have a small level or expansion requiriment so new players actually need to commit a bit to playing the game (but not so much), something like level 5 or 2 expansions to unlock the quests and start being able to get tickets to mint auctioneer items.

>This part was sent by GodIsGreat on discord:
>To create balance between lower and higher level players.
>Quests given can be based on amount of maximum plots/trees/patches/mines that specific player has.

>Example:
>Player 1 (10 plots)
>Quest: Harvest 1 round of cauliflower

>Player 2 (20 plots)
>Quest: Harvest 1 round of cauliflower

>This way. It will be minimal effort when implementing the solution as programmer.
(Because amount of plots/trees/patches/mines is available thru code already)

Another way to get tickets is with the board, that after you click on it you can reedeem one ticket a day, and after logging in for 7 days (in a row or not) the player gets a bonus 3 tickets.

- Now that we know how the player can get tickets, we go to a few options on how the auction can work:

In both of them, when the player wants to mint, he has to bid an amount of tickets that he is willing to give for that SFT/NFT

The first option is that the players that bid more tickets would be the ones that would recieve the NFT/SFT in order of who bid the most until the supply is reached.

The second option is that 50% of the supply would go to those who bid more tickets, while the other 50% of the supply would be drawn in a way that each ticket the player has bid= 1 entry (but he can't win more than once obviously)

While this system is probably hard to add to the game, I think this would be the best way to make it fair for everyone and also reduce frustration on minting items on the auctioneer.

Also a few things to add:

The tickets would not be able to be traded with other players and also would not be refunded if you did not get the NFT/SFT (the resources used would be refunded tho).

**Comment by IvanSkazkin** [comment]
I don't like relying on number generation either. Adding a quest will help active players get nft for their efforts.

**Comment by vitt0** [comment]
I think a system like this looks a lot better that one that is more RNG based, the option with 50% supply for highers bids and the other 50% by RNG sound like a good compromise, the only thing that I would change is to make even easier the examples that you give for the quest, something like:

1) Chop 1 tree
2) Grow 20 cauliflowers
3) Mine 1 gold

I think that this system should include as many player as it's possible, so the ones that still don't have a big farm can still complete them without making a big effort, maybe for someone with a +11 expansions farm 30 cauli dosen't sound as much, but for a newer player who only has 15-25 plots it can be, same thing with the gold, maybe they only have 1 gold in their land.

Also maybe it would be better is the quest that everybody receives daily are diferent for each players, just like the grub shop ones 😄

**Reply by Telknub** [reply]
If this system is chosen we will have a lot of different quests and they could be balanced before releasing the system.

**Comment by poteitouweb3** [comment]
I believe that a good option is for the ticket to work as a kind of advantage in mint. It would form a leaderboard where the mint would be released according to its position. E.g.

Top 1-50: Will be released at 6:00 PM
Top 50-70: Will be released at 6:10 PM
Top 70-150: Will be released at 6:20 PM
And so on...

In other words, the ticket would not guarantee the mint but it would give you the privilege of being able to mint in advance.

**Reply by Telknub** [reply]
This is a good idea, but I think it should made in a different way. After the results, you have a day to mint if you are eligible. Than if someone that could mint did not mint, the NFT would be open for anyone to get, as a last opportunity. 

If a player was eligible to mint but did not mint he would get the resources back after 24h, and the player who was not eligible but got because of the time limit would get the nft in the same way we get it in the current system.

**Comment by LucasAG95** [comment]
I believe this is the best way, as it is fair for everyone

**Comment by xSubZero** [comment]
As I had commented in another thread, I don't like the idea of RNG and this method, it would be much more fair and accessible FOR ALL, without depending solely on your luck.

It is by far the best way to solve the problem.

**Reply by IvanSkazkin** [reply]
Agree

**Reply by rasheed19871** [reply]
i agree as well thus is for better system

**Comment by blastaman** [comment]
This is an amazing idea. Nobody truely like RNG and more that is implemented the more players will leave.

**Comment by PedroGrochibro** [comment]
I think this way will encourage ppl to use multi account to get more chance in the lottery.
Not a good idea in my opinion

**Reply by Telknub** [reply]
How would this encourage multi account? This would actually discourage multi accounting since players would need more work on each island and also lock their resources for a day. It does not change much from what we have now, since if someone wants to try to multiaccount and mint various NFTs he still can right now. 

This system can have multi account measures like a way to detect if there is like a lot of mints in the same household in the 24h  wait period and this household is banned from the mint.

Multi account should not be an issue since they are banned and the game has a bunch of anti botting and anti multi accounting measures.

**Reply by Telknub** [reply]
If you don't like this system you will like even less the system Adam is proposing.

In this system you can still guarantee an NFT by bidding a lot of tickets, Adam's proposal there is no guarantee in any way, just more chances...

**Reply by SaYaNiTo** [reply]
I don t like rng, but i m use to deal with it daily. If some requirements are taken to abort multi account get in lottery massively , why not. Someone said it would be fair for everyone. I m not ok with this. It will be unfair for the top legit players that tryhard the game and will have same chances as a newbie ... Point of view depend wich side of the balance you are.

**Reply by Telknub** [reply]
@SaYaNiTo the point in being fair for everyone is meant to everyone who has the resources to mint an item to have the same chance as other that also can mint that item. There will still be items skewed towards top players and items that are more accessible to everyone. The problem is that these items that are more accessible to everyone always causes trouble and lots of problems. This system would algo give a chance to new players get rarer items, a common complaint beetween lots of new players.

If you are a top player there is no reason to think it is unfair since you should always be able to get lots of tickets easly.

**Reply by SaYaNiTo** [reply]
@Telknub it reminds me so many p2e project. When lottery comes into it s the start of the end, or it was already dead. Hope thisonecould be a counter example.
This decision really need to take in consideration the regular players opinion, cause if most of them leave ... don t need to make a draw.
To stay optimistic, this system could be interesting if making it the right way

**Reply by Telknub** [reply]
@SaYaNiTo the whole point of this suggestion is to reduce RNG and frustration while fixing the minting problems, so I don't think this would make players leave. Even tho one of the option have a draw, in the suggestion 50% would still go to the highest bidders. In my opinion it should be 100% to highest bidders, but I made it a bit more open so people can say what they like/dislike about the options presented and the devs can choose the better option if they like to implement this idea.

**Reply by blastaman** [reply]
I think this proposal is the best idea put forward to fix/improve the minting issues. The alternative Adam is talking about is a completely random lottery… each player can enter once and have a ‘equal chance’ to mint. Maybe I’m wrong but this system would encourage multi accounting, because if you entered 10 times with 10 accounts you have 10x the chance of winning/minting.

But with Telknub’s suggestion multi accounting becomes harder, and less incentivised, as the more effort you put into your one account the more chances you have to win/mint.

**Reply by SaYaNiTo** [reply]
It could be a good idea to create a map or an island (a bit like the rabbit for Chinese new year) which would be fair for all and on which no nft and skill would work.
This island would be there only to have tickets to perform a specific mint.
Tickets would be limited and difficult to obtain.
So whether the players are new or old, all would have same equity for mints (required they have the necessary materials).
The island would be available for planting or pickaxes only when there is a mint to be made.
Crops, fruits or rocks created on this island would only be usable for mints and won t be withdrawable.

**Reply by SaYaNiTo** [reply]
This is an idea french community discussed about earlier. Let s talk about it , first thing is would it be possible and not too constraining for devs ?

**Comment by celpower** [comment]
The only thing that didn't make sense to me was easier tasks gets more tickets, I would personally always do the easier one. :-) I am still trying to wrap my head around the idea of totally changing the mint as is, but I am trying to be as open as possible to ideas.

**Reply by Telknub** [reply]
The idea is that even if you don't do the harder tasks (because you are new to the game or because you did not feel like it) you would not fall too much behind other players in the amount of tickets, making those who do all quests have more tickets but not by a large margin. You would be able to do all 3 quests every day, so it is up to the player. This way every player can compete on a chance to get the SFT.

The minting system has to change, I am not saying it has to be exactly like I am saying (even tho I would want it like this lol) but we need to find solutions to make it more fair for everyone and also to reduce frustration on minting items because of bugs or errors.

**Reply by blastaman** [reply]
I like this idea a lot too. I see how it can work and level the playing field for all players. The idea is that advanced players still gain an advantage by being able to complete all tasks, but the advantage is not huge… and it based on commitment to your farm.

Say you had the tiered tasks and easy gives you 5 tickets, Med gives you 3 and hard gives you 1 extra ticket. After ten days a player that could only complete the Easy and Med tasks would have 80 tickets, but a more advance player that could complete them all would only have 90 tickets…

Still people will say this is unfair, but then you have to bid with your tickets for the mint right? So advance player can bid 90 max (but might chose to save 10 for advantage in next mint) and mid player can bid 80.

After a few mints, some low/mid level players who missed out on earlier mint would have more tickets than a advance player, giving them a chance to out bid them on the next mint.

So this system gives all players a shot at a he different mints… with as little RNG as possible…. Love it!

**Reply by celpower** [reply]
I totally missed the part you can do more than one task!

**Comment by adamhannigan** [comment]
Nice writeup! @Telknub  During the community chat yesterday sentiment definitely was not to use RNG and rely on some quests.

Building custom quests takes time and is error prone unfortunately - ideally we want to use something that has been tested and is a core gameplay loop.

What are your thoughts around using the Grub Shop as the quests? Every meal you fulfill you get SFL + a Lottery ticket.

This plays nicely with our current game loop and rewards people who are completing the entire shop. Naturally this means people at higher levels can earn more tickets as they have access to more food.

**Reply by Telknub** [reply]
I think that this is a pretty good idea on how to get tickets to already be working in a fast way, and then later the system can be improved and make it even better. The main problem of this idea as a whole is that it would take time to implement 🤣 

I really like the idea behind the strategy of how many tickets player can bid, and the way the SFTs are minted need to be without RNG (at least half of the supply imo). This would reduce a lot of the frustration players have with the current system.

**Reply by IvanSkazkin** [reply]
It already sounds better. Food tickets, great.

Take your time Adam. We'll wait for quests for tickets. It's better than doing a random generation that people will run away from.

**Reply by Telknub** [reply]
If the team goes with the grub shop approach each item on the grub shop has to give the same amount of tickets in my opinion (1 for each one of them). 

But there is a caveat. The grub shop gives different items for different people, so sometimes someone's grub shop could be easier than others and this can lead to some complaints from players (also players  that do not have chickens or fruit might feel they are falling behind because of it). It will still be a lot better than the current system, but it won't be perfect.

**Reply by ReC82** [reply]
> If the team goes with the grub shop approach each item on the grub shop has to give the same amount of tickets in my opinion (1 for each one of them).
> 
> But there is a caveat. The grub shop gives different items for different people, so sometimes someone's grub shop could be easier than others and this can lead to some complaints from players (also players that do not have chickens or fruit might feel they are falling behind because of it). It will still be a lot better than the current system, but it won't be perfect.


There it is, the random evil guy ;). joking.
I really like the idea btw.
And what about recipe not doable by new players ?  Like involving eggs ? cakes ? etc.

"This plays nicely with our current game loop and rewards people who are completing the entire shop. Naturally this means people at higher levels can earn more tickets as they have access to more food."

And the ones who completes everything gets an extra ticket ^^.

**Reply by IvanSkazkin** [reply]
How about giving away tickets for cooking?
For example, 10% chance to get a ticket for cooking food. It will make people play a lot

**Reply by Telknub** [reply]
@IvanSkazkin this would be bad because it would be a random thing

it would also incentivize people to do the faster foods and make it not very balanced
the grub shop is the best approach to this

**Reply by IvanSkazkin** [reply]
> @IvanSkazkinэто было бы плохо, потому что это была бы случайная вещь
> 
> это также будет стимулировать людей есть более быструю еду и сделает ее не очень сбалансированной , магазин личинок - лучший подход к этому

Agree you are right.

**Reply by celpower** [reply]
I like the idea of the grubshop as a lot of players don't like to cook food to sell food in GrrubShop. That would force people to not only plant specific crops and be open to grow different crops or buy from other players.

**Comment by adamhannigan** [comment]
Also, for those that haven't read the PR yet. Yesterday the approach shifted from RNG to lottery tickets - we are all trying to decide what the fairest way to give out lottery tickets would be.

https://github.com/sunflower-land/sunflower-land/pull/2122

Any thoughts appreciated ❤️ 

<img width="945" alt="Screen Shot 2023-02-01 at 8 41 09 am" src="https://user-images.githubusercontent.com/11745561/215889756-65e61b2e-3c29-45ec-9027-48df2b800596.png">

**Comment by Aquila199** [comment]
having just read the new update on this matter

the only issue i can see is the part where bid times are used this should be random or something as timezones will be a massive issue and make it unfair again

**Comment by VZozul** [comment]
Today, a lot was written in the chat, about the same food for levels, about the fact that everyone has different levels and a different number of plots and what's unfair is that whoever has less and who can't cook everything in the Grub Shop will get fewer tickets. But what about the motivation to raise the level, make expansions, again, if you do this, then multi-accounts will make the minimum level to collect tickets. On the contrary, there will be motivation for newcomers to grow up faster in order to participate in auctions, collect more tickets and take away some multi-accounts. The same goes for the fact that some with less expansions have less plots. Let them grow, discover new lands, get more tickets. This is a natural growth process. 
The only question left is with boosts, and it's true that those who have a lot of boosts will be far ahead. Because for example with Mysterious Parsnip you will get much more tickets. Well, on the one hand, you paid for this boost, you have an advantage, on the other hand, everything will again come down to money and take away the excitement from others, it’s important how much time you invest, for honesty. So here you need to find the right balance. Although even for example with the same Mysterious Parsnip you already have the boosts that you used before, these boosts do not necessarily have to affect the tickets as well.
+ 
And about the same food to be honest, why? For 1 food, for example, 1 ticket and that's it, the rest of the tickets are from the harvest ... Again, from what I wrote above, if you want to get more tickets, grow to be able to fill the entire Grub Shop, expand to have more plots. Yes, and the random Grub Shop still adds a little RNG, but if you can cook everything, then this random will not affect you, since you still get 1 ticket for 1 meal. And it's okay that at first there will not be a big difference in the number of tickets, except for those who have not yet grown up and have not made all the expansions. The gist is "This means you must strategies and plan accordingly when you want to use the tickets. The goal is to bid the least amount of tickets possible, ensuring that you save tickets for other seasonal items or to swap for SFL." from docs
+
+
After some discussions in the chat, we came up with an idea how to make the game fair and fair mint SFTs for all levels.
First option

![INR](https://user-images.githubusercontent.com/103659539/221601106-d029ba5e-5c4f-47ae-b911-7c002f6d3a66.jpg)

The second option is to make each level range its own craft and its own auction with different boost values.
For example
Level 1-20 - +0.1
Level 20-40 - +0.25
Level 40-50 - +0.5
In this case, the distribution of tickets as in the screen. Lower levels will be able to receive the same number of tickets as the maximum levels. But they will be able to mint SFTs in their range but with a lower boost coefficient. Roughly speaking, break up by Tiers. With mint limits with their own level ranges, both minimum and maximum. In a word, give small levels the opportunity to mint something that will give a boost, but much less than that of larger levels. 
It's even better to tie it to expansions, since they already require a minimum level. Bumpkins can be transferred with different levels in order to remove at least a little multi-accounts, let them expand.
And in order to avoid multi-accounts in order to get all Tiers, make sure that the same SFT but different levels of efficiency do not stack.

There are many options, starting from justice, you can figure out how to improve this system so that every player, even the newest one, can participate and enjoy the game and boosts.

**Comment by future2back** [comment]
How can a new player have any chance against an old player?

This system only makes it difficult for minors to access NFTs and makes it easier for big ones, once again a modification that strengthens a specific group and forgets the players.




Are we still in the beta version to continue doing these tests?  I'm sure this won't work.
